### DISCUSSION TITLE
DataSource error - Cost Analisys

### DISCUSSION THREAD

**Discussion by Andretalyss** [cloudforet-io/community#140] [General]
I'm getting an error when creating dashboards:

```
Cost.analyze (service.py:171) (Error) => Required parameter. (key = data_source_id) 
	error_code = ERROR_REQUIRED_PARAMETER
	status_code = INVALID_ARGUMENT
	message = Required parameter. (key = data_source_id)  
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/spaceone/core/service/service.py", line 122, in _pipeline
    response_or_iterator = func(self, params)
  File "/usr/local/lib/python3.8/site-packages/spaceone/core/service/utils.py", line 41, in wrapped_func
    raise ERROR_REQUIRED_PARAMETER(key=key)
spaceone.core.error.ERROR_REQUIRED_PARAMETER: 
	error_code = ERROR_REQUIRED_PARAMETER
	status_code = INVALID_ARGUMENT
	message = Required parameter. (key = data_source_id)
```

And in the datasource creation panel, it does not allow creation, just an information page.

![Screenshot from 2024-02-01 12-19-38](https://github.com/cloudforet-io/cloudforet/assets/45321557/96b6856f-0d0e-41cb-878f-b488aae41f21)

Can anyone help?

**Comment by choonho** [comment]
@Andretalyss See this example. It is similar with your problem.
I think you did not pass the data_source_id which was created by "spacectl exec register cost_analysis.DataSource -f <parameter files>"

If you want to check your registered data source, "spacectl list cost_analysis.DataSource -c data_source_id,name"

**Comment by choonho** [comment]
@Andretalyss I can guess your problem. You first enable cost_analysis by CLI tool. Sorry we didn't problem frontend to enable cost_analysis.

To enable cost_analysis,
1. Register cost_analysis plugin (by CLI)
2. Enable cost_analysis plugin (by CLI)
3. Sync data source (by CLI)

https://github.com/orgs/cloudforet-io/discussions/139, this is one of example for file data source plugin.

If you want quick response, visit our discord channel.  https://discord.gg/7ExpTmA6TE
